plugins/quota/init.go

```diff
@@ -0,0 +1,110 @@
+package quota
+
+import (
+	"github.com/gluster/glusterd2/glusterd2/servers/rest/route"
+	"github.com/prashanthpai/sunrpc"
+)
+
+// Plugin is a structure which implements GlusterdPlugin interface
+type Plugin struct {
+}
+
+// Name returns name of plugin
+func (p *Plugin) Name() string {
+	return "quota"
+}
+
+// SunRPCProgram returns sunrpc program to register with Glusterd
+func (p *Plugin) SunRPCProgram() sunrpc.Program {
+	return nil
+}
+
+// RestRoutes returns list of REST API routes to register with Glusterd
+func (p *Plugin) RestRoutes() route.Routes {
+	return route.Routes{
+		route.Route{
+			Name:        "QuotaHelp",
+			Method:      "GET",
+			Pattern:     "/quota/help",
+			Version:     1,
+			HandlerFunc: quotaHelpHandler},
+		route.Route{
+			Name:        "QuotaEnable",
+			Method:      "POST",
+			Pattern:     "/quota/{volname}/enable",
+			Version:     1,
+			HandlerFunc: quotaEnableHandler},
+		route.Route{
+			Name:        "QuotaDisable",
+			Method:      "POST",
+			Pattern:     "/quota/{volname}/disable",
+			Version:     1,
+			HandlerFunc: quotaDisableHandler},
+		route.Route{
+			Name:        "QuotaList",
+			Method:      "GET",
+			Pattern:     "/quota/{volname}/list",
+			Version:     1,
+			HandlerFunc: quotaListHandler},
+		route.Route{
+			Name:        "QuotaListObjects",
+			Method:      "GET",
+			Pattern:     "/quota/{volname}/list-objects",
+			Version:     1,
+			HandlerFunc: quotaListObjectsHandler},
+		route.Route{
+			Name:        "QuotaLimitUsage",
+			Method:      "POST",
+			Pattern:     "/quota/{volname}/limit-usage",
+			Version:     1,
+			HandlerFunc: quotaLimitUsageHandler},
+		route.Route{
+			Name:        "QuotaLimitObjects",
+			Method:      "POST",
+			Pattern:     "/quota/{volname}/limit-objects",
+			Version:     1,
+			HandlerFunc: quotaLimitObjectsHandler},
+		route.Route{
+			Name:        "QuotaRemove",
+			Method:      "POST",
+			Pattern:     "/quota/{volname}/remove",
+			Version:     1,
+			HandlerFunc: quotaRemoveHandler},
+		route.Route{
+			Name:        "QuotaRemoveObjects",
+			Method:      "POST",
+			Pattern:     "/quota/{volname}/remove-objects",
+			Version:     1,
+			HandlerFunc: quotaRemoveObjectsHandler},
+		route.Route{
+			Name:        "QuotaDefaultSoftLimit",
+			Method:      "POST",
+			Pattern:     "/quota/{volname}/default-soft-limit",
```

Comment:
We can have single route for all the configurations

    POST /quota/{volname}/config

Request body:

    {
        "default-soft-limit": 70
    }

limit-usage, default-soft-limit, alert-time, soft-timeout, hard-timeout

and reset API

    DELETE /quota/{volname}/config

Reply:
All options that are [part of quota xlator](https://github.com/gluster/glusterfs/blob/master/xlators/features/quota/src/quota.c#L5263) can be a normal volume set option.

Reply:
@aravindavk , @prashanthpai, 

I don't see an issue with suggestion by @prashanthpai as long as the necessary validations can be done in volume set path (volume is started , quota is enabled).

The glustercli would have to be mapped to the volume set route to maintain backward compatibility. So as far as glustercli is concerned, the same volume option would be visible in 2 ways. 
eg:
> gluster_cli quota <volname> soft-timeout <value>
> gluster volume set <volname> quota.soft-timeout <value>
Is this ok ?  I think yes!

If we agree, will also have to refresh the patch to make the options settable again
https://review.gluster.org/#/c/18743/4/xlators/features/quota/src/quota.c

Reply:
We need to have an answer to https://github.com/gluster/glusterd2/issues/501 sooner than later to get to the decision on the above.

Reply:
Discussion notes:
with @aravindavk , @harigowtham

Decided to go with options route for setting quota config (as suggested by @prashanthpai).
#501 would be addressed to have validators which quota can also make use of.